src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -22,58 +21,172 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
         public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, Logger logger, CancellationToken cancellationToken)
         {
-            FeatureFlag featureFlag;
-            try
-            {
-                 featureFlag = JsonSerializer.Deserialize<FeatureFlag>(setting.Value);
-            }
-            catch (JsonException e)
-            {
-                throw new FormatException(setting.Key, e);
-            }
-
             var keyValues = new List<KeyValuePair<string, string>>();
 
-            if (featureFlag.Enabled)
+            try
             {
-                //if (featureFlag.Conditions?.ClientFilters == null)
-                if (featureFlag.Conditions?.ClientFilters == null || !featureFlag.Conditions.ClientFilters.Any()) // workaround since we are not yet setting client filters to null
-                {
-                    //
-                    // Always on
-                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}", true.ToString()));
-                }
-                else
+                using (JsonDocument document = JsonDocument.Parse(setting.Value))
```

Comment:
In the thread to move away from JsonSerializer the suggestion was to move to Utf8JsonReader

Reply:
When I first was looking at how to use utf8jsonreader, it seemed like the logic would get pretty verbose for the amount of properties we have to add, so I thought it'd be easier to read/follow by using JsonDocument. It adds a little bit of abstraction but uses utf8jsonreader in its implementation. I wasn't sure what the benefits were, but if it's just for consistency reasons since we use it elsewhere then I can change it.

Reply:
It's to have a consistent approach to serialization/deserialization across the system.